ams/optimizer/src/main/java/com/netease/arctic/optimizer/KubernetesOptimizerContainer.java

```diff
@@ -0,0 +1,89 @@
+
+
+  @Override
+  protected Map<String, String> doScaleOut(String startUpArgs) {
+    try {
+      String[] cmd = { startUpArgs};
+      LOG.info("Starting local optimizer using command : {}", startUpArgs);
+      Config config = Config.fromKubeconfig(KUBECONFIG_CONTENTS);
+      KubernetesClient client = new KubernetesClientBuilder().withConfig(config).build();
+      K8sUtil.runJob("default","amoro-optimizer",
+          client, "registry.cn-hangzhou.aliyuncs.com/udh/amoro-optimizer:0.6.0",String.join(" ",cmd),LOG);
+      return Collections.emptyMap();
+    } catch (Exception e) {
+      throw new RuntimeException("Failed to scale out optimizer.", e);
+    }
+  }
+
+  @Override
+  protected String buildOptimizerStartupArgsString(Resource resource) {
+    long memoryPerThread = Long.parseLong(PropertyUtil.checkAndGetProperty(resource.getProperties(),
+        JOB_MEMORY_PROPERTY));
+    long memory = memoryPerThread * resource.getThreadCount();
+    return String.format("%s/bin/localOptimize.sh %s %s", "/usr/local/amoro", memory,
+        super.buildOptimizerStartupArgsString(resource));
+  }
```

Comment:
I believe that implementing requestResource directly in K8S mode may be more reasonable. AbstractResourceContainer can even be refactored. 

In my opinion, the abstract methods designed in AbstractResourceContainer are not reasonable at present. those methods are designed to manage the Optimizer through shell commands.